src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -661,6 +669,57 @@ private void validateRenewalInfo(CompositePurchaseOrder compPO) {
       .thenApply(v -> compPO);
   }
 
+  private CompletableFuture<Void> validateFundDistributionTotal(CompositePurchaseOrder compPO) {
+    if (compPO.getWorkflowStatus() == WorkflowStatus.OPEN && compPO.getTotalEstimatedPrice() != null) {
+      Double totalEstimatedPrice = compPO.getTotalEstimatedPrice();
+      Double remainingPercent = 100.0;
+
+      for (CompositePoLine cPoLine : compPO.getCompositePoLines()) {
+        String currency = cPoLine.getCost().getCurrency();
+        MonetaryAmount remainingAmount = Money.of(totalEstimatedPrice, currency);
+
+        for (FundDistribution fundDistribution : cPoLine.getFundDistribution()) {
+          DistributionType dType = fundDistribution.getDistributionType();
+
+          if (dType == DistributionType.PERCENTAGE) {
+            Double percentValue = fundDistribution.getValue();
+            remainingPercent -= percentValue;
+
+            MonetaryAmount totalEstimatedPriceMoney = Money.of(totalEstimatedPrice, currency);
+            /**
+             * calculate remaining amount to carry forward, required if there are more fund distributions with percentage and amount
+             * currentAmount = totalEstimatedPrice * value/100;
+             * remainingAmount = totalEstimatedPrice - currentAmount;
+             */
+            MonetaryAmount currentAmount = totalEstimatedPriceMoney.with(MonetaryOperators.percent(percentValue));
+            remainingAmount = remainingAmount.subtract(currentAmount);
+
+            if (remainingPercent < 0) {
```

Comment:
Description says fundDistributions must add upto 100% of total.  Doesn't that mean that their sum should be equal to the total value?

Reply:
I followed UI. In UI `Remaining amount to be distributed` in Fund distribution is set to total and subtracted from the total at runtime. Therefore, the total equals `0` when the sum equals the total.

Reply:
@craigmcnally could you clarify which of these is true: how it's described in the ticket or how it's implemented in the UI

Reply:
It almost sounds like you're both saying the same thing...

If we have a total of $100

Valid:
* 50%, 50%
* $25, $75
* $50, 50%

Invalid:
* 50% (need to distribute the other 50% to one or more funds)
* 33%, 33%, 33% (only adds up to 99%)
* $50, 25%, only adds up to $75)

Reply:
In that case logic has to be updated